Unused private field generates warning in clang

**Describe the bug**
in clang debug

```
../TPL/vt/src/vt/worker/worker_stdthread.h:88:19: error: private field 'num_thds_' is not used [-Werror,-Wunused-private-field]
```

--- a/src/vt/worker/worker_stdthread.cc
+++ b/src/vt/worker/worker_stdthread.cc
@@ -59,7 +59,8 @@
 namespace vt { namespace worker {
 
 StdThreadWorker::StdThreadWorker(
-  WorkerIDType const& in_worker_id_, WorkerFinishedFnType finished_fn
+  WorkerIDType const& in_worker_id_, WorkerCountType const&,
+  WorkerFinishedFnType finished_fn
 ) : worker_id_(in_worker_id_), finished_fn_(finished_fn)
 { }
 

--- a/src/vt/worker/worker_stdthread.cc
+++ b/src/vt/worker/worker_stdthread.cc
@@ -59,9 +59,8 @@
 namespace vt { namespace worker {
 
 StdThreadWorker::StdThreadWorker(
-  WorkerIDType const& in_worker_id_, WorkerCountType const& in_num_thds,
-  WorkerFinishedFnType finished_fn
-) : worker_id_(in_worker_id_), num_thds_(in_num_thds), finished_fn_(finished_fn)
+  WorkerIDType const& in_worker_id_, WorkerFinishedFnType finished_fn
+) : worker_id_(in_worker_id_), finished_fn_(finished_fn)
 { }
 
 void StdThreadWorker::enqueue(WorkUnitType const& work_unit) {

--- a/src/vt/worker/worker_stdthread.h
+++ b/src/vt/worker/worker_stdthread.h
@@ -85,7 +85,6 @@ private:
 private:
   std::atomic<bool> should_terminate_ = {false};
   WorkerIDType worker_id_ = no_worker_id;
-  WorkerCountType num_thds_ = no_workers;
   WorkUnitContainerType work_queue_;
   ThreadPtrType thd_ = nullptr;
   WorkerFinishedFnType finished_fn_ = nullptr;

--- a/src/vt/worker/worker_stdthread.h
+++ b/src/vt/worker/worker_stdthread.h
@@ -67,7 +67,8 @@ struct StdThreadWorker {
   using WorkUnitContainerType = util::container::ConcurrentDeque<WorkUnitType>;
 
   StdThreadWorker(
-    WorkerIDType const& in_worker_id_, WorkerFinishedFnType finished_fn
+    WorkerIDType const& in_worker_id_, WorkerCountType const&,
+    WorkerFinishedFnType finished_fn
   );
   StdThreadWorker(StdThreadWorker const&) = delete;
 

--- a/src/vt/worker/worker_stdthread.h
+++ b/src/vt/worker/worker_stdthread.h
@@ -67,8 +67,7 @@ struct StdThreadWorker {
   using WorkUnitContainerType = util::container::ConcurrentDeque<WorkUnitType>;
 
   StdThreadWorker(
-    WorkerIDType const& in_worker_id_, WorkerCountType const& in_num_thds,
-    WorkerFinishedFnType finished_fn
+    WorkerIDType const& in_worker_id_, WorkerFinishedFnType finished_fn
   );
   StdThreadWorker(StdThreadWorker const&) = delete;
 

